napari/layers/transforms.py

```diff
@@ -60,12 +72,67 @@ def set_pad(self, axes: Sequence[int]) -> 'Transform':
         """
         raise NotImplementedError('Cannot subset arbitrary transforms.')
 
+
+class TransformChain(ListModel, Transform):
+    def __init__(self, transforms=[]):
+        super().__init__(
+            basetype=Transform,
+            iterable=transforms,
+            lookup={str: lambda q, e: q == e.name},
+        )
+
+    def __call__(self, coords):
+        return tz.pipe(coords, *self)
+
+    def __newlike__(self, iterable):
+        return ListModel(self._basetype, iterable, self._lookup)
+
     @property
-    def inverse(self) -> 'Transform':
-        if self._inverse_func is not None:
-            return Transform(self._inverse_func, self.func)
+    def inverse(self) -> 'TransformChain':
+        """Return the inverse transform chain."""
+        return TransformChain([tf.inverse for tf in self[::-1]])
+
+    @property
+    def composite(self) -> 'Transform':
+        """Return a composite of the transform chain."""
+        if len(self) == 0:
+            return None
+        if len(self) == 1:
+            return self[0]
         else:
-            raise ValueError('Inverse function was not provided.')
+            return tz.pipe(self[0], *[tf.compose for tf in self[1:]])
+
+    def set_slice(self, axes: Sequence[int]) -> 'TransformChain':
+        """Return a transform chain subset to the visible dimensions.
+
+        Parameters
+        ----------
+        axes : Sequence[int]
+            Axes to subset the current transform chain with.
+
+        Returns
+        -------
+        TransformChain
+            Resulting transform chain.
+        """
+        return TransformChain([tf.set_slice(axes) for tf in self])
+
+    def set_pad(self, axes: Sequence[int]) -> 'Transform':
```

Comment:
Ah! Can we rename this `expand_dims`? I was totally confused about pad because padding in NumPy land is of existing dims. `expand_dims` is exactly the analogous function in NumPy.

Reply:
will do

Reply:
done